packages/x-telemetry/src/config.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable no-underscore-dangle */
+import { ponyfillGlobal } from '@mui/utils';
+
+interface TelemetryEnv {
+  NODE_ENV?: string;
+  IS_COLLECTING: boolean | null;
+  DEBUG: boolean;
+}
+
+let cachedEnv: TelemetryEnv | null = null;
+
+function getMuiXEnabledEnvs() {
+  return (
+    process.env.MUI_X_TELEMETRY_ENABLED ||
+    process.env.REACT_APP_MUI_X_TELEMETRY_ENABLED ||
+    process.env.NEXT_PUBLIC_MUI_X_TELEMETRY_ENABLED ||
+    process.env.GATSBY_MUI_X_TELEMETRY_ENABLED ||
+    process.env.PUBLIC_MUI_X_TELEMETRY_ENABLED
+  );
```

Comment:
To support all prefixes
```suggestion
  return (Object.entries(process.env).find(([key, _]) => key.endsWith('MUI_X_TELEMETRY_ENABLED')) || [])[1];
```
Also for the disabled var